edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/DependencyManager.cs

```diff
@@ -123,7 +127,11 @@ void RegisterMqttModule(ContainerBuilder builder, (bool isEnabled, bool usePersi
             bool clientCertAuthEnabled = this.configuration.GetValue(Constants.ConfigKey.EdgeHubClientCertAuthEnabled, false);
 
             IConfiguration mqttSettingsConfiguration = this.configuration.GetSection("mqttSettings");
-            builder.RegisterModule(new MqttModule(mqttSettingsConfiguration, topics, this.serverCertificate, storeAndForward.isEnabled, clientCertAuthEnabled, optimizeForPerformance, this.sslProtocols));
+
+            if (mqttSettingsConfiguration.GetValue("enabled", false))
+            {

```

Comment:
in 'GetEdgeHubProtocolHeadAsync' the legacy mqtt depends on 2 flags (mqttSettingsConfiguration and experimental/brokerEnabled) while here the registration happens based only on mqttSettingsCongfiguration. Doesn't seem a problem as it cannot happen the legacy mqtt is not registered but then enabled, but is it worth to use the same logic here to make it consistent?

Reply:
yea good point